### DISCUSSION TITLE
OpenPNM dev version import error

### DISCUSSION THREAD

**Discussion by minzhangut** [PMEAL/OpenPNM#2377] [Q&A] [answered]
I did the following installations:

1. conda install -c conda-forge porespy
2. conda uninstall openpnm
3. pip install git+https://github.com/PMEAL/OpenPNM.git@dev

I did not see errors when I did the installations.

But it has errors when I tried to "import openpnm":
ImportError: cannot import name 'pore_size' from partially initialized module 'openpnm.models.geometry'

**Comment by ma-sadeghi** [comment, marked as answer]
That's a bug, so I'm converting this into a proper issue, will push a fix tonight.

**Reply by ma-sadeghi** [reply]
@minzhangut It's now fixed, give it a try please!

**Reply by minzhangut** [reply]
4. pip uninstall openpnm
5. pip install git+https://github.com/PMEAL/OpenPNM.git@dev

I did not have issue with "import openpnm", but it says "ModuleNotFoundError: No module named 'porespy'" when I tried to "import porespy".

So I repeated the steps again.

6. conda install -c conda-forge porespy
7. conda uninstall openpnm
8. pip install git+https://github.com/PMEAL/OpenPNM.git@dev

Then I got this error during installation:
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '***/pytool_v3/lib/python3.10/site-packages/numpy-1.21.6.dist-info/METADATA'

**Reply by minzhangut** [reply]
I started from scratch:

1. conda install -c conda-forge porespy
2. conda uninstall openpnm
3. pip install git+https://github.com/PMEAL/OpenPNM.git@dev

I am having the same issue:
  ```
        In [1]: import porespy
        ---------------------------------------------------------------------------
        ModuleNotFoundError                       Traceback (most recent call last)
        Input In [1], in <cell line: 1>()
        ----> 1 import porespy
        
        ModuleNotFoundError: No module named 'porespy'
        
        In [2]: import openpnm
        
        In [3]: openpnm.__version__
        Out[3]: '2.8.2.dev1116'
```

**Reply by ma-sadeghi** [reply]
Why not installing `porespy` the same way you install `openpnm`?

```bash
pip install git+https://github.com/PMEAL/OpenPNM.git@dev
pip install git+https://github.com/PMEAL/porespy.git@dev
```

It should work fine (I tried it on Linux and it worked).

**Reply by minzhangut** [reply]
I am having issue when I tried to install porespy.

```
error: command '/bin/gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

Ã Encountered error while trying to install package.
â°â> pyfastnoisesimd

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.

```

**Reply by ma-sadeghi** [reply]
1. Are you on Windows?
2. If not, could you repeat the process in a fresh virtual env?
```
conda create -n testenv python=3.9
conda activate testenv
pip install git+https://github.com/PMEAL/OpenPNM.git@dev
pip install git+https://github.com/PMEAL/porespy.git@dev
```

**Reply by minzhangut** [reply]
It worked. Thanks!
